emacs/coalton-mode.el

```diff
@@ -0,0 +1,269 @@
+;;; coalton-mode.el --- Major mode for working with Coalton -*- lexical-binding: t; -*-
+;;
+;; URL: http://github.com/coalton-lang/coaltom
+;; Keywords: languages coalton lisp
+;; Version: 1.0.0
+;; Package-Requires: ((emacs "29.1"))
+;;
+;; This file contains functions for in-Emacs structural operations on
+;; Coalton code, including syntax highlighting, indentation and
+;; navigation, and command integration with the in-CL operations
+;; defined in `slime-coalton.el'.
+
+(require 'treesit)
+(require 'lisp-mnt)
+(require 'slime)
+(require 'slime-coalton)
+
+(add-to-list 'slime-contribs 'slime-coalton)
+
+(defconst coalton-mode-version
+  (eval-when-compile
+    (lm-version (or load-file-name buffer-file-name))))
+
+(defvar coalton-ts-repo
```

Comment:
can we put this in a self-contained place?

Reply:
this is now a URL for coalton-lang/tree-sitter-coalton